lib/options.go

```diff
@@ -44,6 +45,79 @@ const DefaultScenarioName = "default"
 // nolint: gochecknoglobals
 var DefaultSummaryTrendStats = []string{"avg", "min", "med", "max", "p(90)", "p(95)"}
 
+type DNSConfig struct {
+	TTL      null.String     `json:"ttl"`
+	Strategy NullDNSStrategy `json:"strategy"`
+}
+
+//go:generate enumer -type=DNSStrategy -transform=kebab -trimprefix DNS -output dns_strategy_gen.go
+type DNSStrategy uint8
+
+type NullDNSStrategy struct {
+	DNSStrategy
+	Valid bool
+}
+
+const (
+	DNSFirst DNSStrategy = iota + 1
+	DNSRoundRobin
+	DNSRandom
+)
+
+// nolint: gochecknoglobals
+var DefaultDNSConfig = DNSConfig{
+	TTL:      null.StringFrom("inf"),
+	Strategy: NullDNSStrategy{DNSFirst, true},
+}
+
```

Comment:
I would prefer this to be a getter/generator/construct like ... most other such ones which just returns a new instance.

I do worry a lot with this kind of pointer heavy structures that someone will copy the value but the value will have a pointer inside and then they will just change the value to which the pointer points instead of the pointer. 
I don't think this happens here (or is even possible), but in the future, it might be and those kind of bugs are hard to see usually

Reply:
Resolved by 7578a411 ?